### PR TITLE
PLT-661:Added Cluster profile support for cluster group resource

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/resource_cluster_group_test.go [hunk 1]
```diff
@@ -32,6 +32,11 @@ func prepareClusterGroupTestData() *schema.ResourceData {
 			"host_dns":    "https://test.dev.spectro.com",
 		},
 	})
+	d.Set("cluster_profile", []map[string]interface{}{
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group_test.go:35]: @SivaanandM please handle set error in the test too.

File: spectrocloud/resource_cluster_group.go [hunk 2]
```diff
@@ -117,9 +119,8 @@ func resourceClusterGroupCreate(ctx context.Context, d *schema.ResourceData, m i
 
 	// Warning or errors can be collected in a slice type
 	var diags diag.Diagnostics
-
-	cluster := toClusterGroup(d)
 	scope := d.Get("context").(string)
+	cluster := toClusterGroup(c, d)
```
Comment (nikchern) [inline spectrocloud/resource_cluster_group.go:123]: @SivaanandM I haven't found changes for toClusterGroup() is it pushed? 
same for the flatten function and corresponding unit test for flatten function.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_cluster_group_test.go (hunk 1 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group_test.go]: addressed

File: spectrocloud/resource_cluster_group.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_group.go:123]: @nikchern as discussed we are not flattening cluster profiles

**Review by nikchern — APPROVED** [review]

**Pull request merged**